File: backend/src/main/java/com/workstation/modules/task/entity/Task.java
package com.workstation.modules.task.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.workstation.common.entity.BaseEntity;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@TableName("task")
public class Task extends BaseEntity {

    private String title;

    private String description;

    private TaskType taskType;

    private Integer priority;

    /** TODAY 任务的日期；重复任务的重复起始日 */
    private LocalDate planDate;

    /** LONG_TERM 的截止日，可为空 */
    private LocalDate dueDate;

    private RecurrenceType recurrenceType;

    private Integer recurrenceInterval;

    private LocalDate recurrenceEndDate;

    /** 非重复任务的完成状态；重复任务看各自的 task_occurrence */
    private TaskStatus status;

    public boolean isRecurring() {
        return recurrenceType != null && recurrenceType != RecurrenceType.NONE;
    }

    public boolean isLongTerm() {
        return taskType == TaskType.LONG_TERM;
    }

    /**
     * 这条任务在日历上落在哪一天；返回 null 表示它不占日历格子。
     *
     * 长期任务返回 null —— 它是一份待办清单，不是一个「某天要做的事」。
     * 之前给了它日期，结果长期任务混进了今日任务里。
     */
    public LocalDate anchorDate() {
        if (isLongTerm()) {
            return null;
        }
        return planDate;
    }
}

File: frontend/src/utils/task.ts
import type { TaskOccurrenceVO, TaskVO } from '@/types/domain'

/**
 * 把「某天的任务实例」转成编辑表单需要的任务形状。
 *
 * 实例里只有当天那一条的信息，任务的日期/重复规则不在其中，
 * 所以日期取实例所在的那天、重复规则按原样带回——
 * 编辑时必须把这些原值传回去，否则后端会把「字段为空」当成「改成默认值」，
 * 把任务挪到今天、或者把重复任务改成不重复。
 */
export function occurrenceToTask(item: TaskOccurrenceVO): TaskVO {
  return {
    id: item.taskId,
    title: item.title,
    description: item.description,
    taskType: item.taskType,
    priority: item.priority,
    planDate: item.occurDate,
    dueDate: item.taskType === 'LONG_TERM' ? item.occurDate : null,
    recurrenceType: item.recurrenceType,
    recurrenceInterval: 1,
    recurrenceEndDate: null,
    status: item.status,
    recurring: item.recurring,
    anchorDate: item.occurDate,
  }
}
